stock_location_address_purchase/models/purchase.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2018 Creu Blanca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.depends("picking_type_id")
+    def _compute_dest_address_id(self):
+        res = super()._compute_dest_address_id()
+        for po in self:
```

Comment:
Use filtered, better tahn for - if

Reply:
@angelmoya hmm it was actually better before. Before it was one single iteration to the recordset, now it is a full iteration to apply filtered and the another iteration to the resulting recordset.

Could you clarify why it is better from your POV?